hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/security/token/block/BlockTokenSecretManager.java

```diff
@@ -221,12 +225,16 @@ private synchronized void removeExpiredKeys() {
   /**
    * Set block keys, only to be used in worker mode
    */
-  public synchronized void addKeys(ExportedBlockKeys exportedKeys)
+  public synchronized void addKeys(ExportedBlockKeys exportedKeys, Optional<InetSocketAddress> nnAddr)
       throws IOException {
     if (isMaster || exportedKeys == null) {
       return;
     }
-    LOG.info("Setting block keys. BlockPool = {} .", blockPoolId);
+    if (nnAddr.isPresent()) {
+      LOG.info("Setting block keys. BlockPool = {} . Namenode Address = {} .", blockPoolId, nnAddr);
```

Comment:
Can we pass the BPServiceActor instead? That way you can print the hostname as well as the state. This will display whether or not it was the active/standby/observer when the request happened. This could be useful debugging information.

Reply:
Discussed and will not go forward with this approach due to BPServiceActor being private